### PR TITLE
feat(googleapi): support setting arbitrary query parameters

### PR DESCRIPTION
**Pull request by codyoss**
Fixes: #550

### CODE REVIEW

**Review by tbpg — COMMENTED** [review]
I think this had come up before with @broady, so will leave review to him.

**Review by tbpg — COMMENTED** [review]
Found https://github.com/GoogleCloudPlatform/golang-samples/pull/1015 for some history.

**Comment by codyoss** [comment]
Yeah I am not sure if there is any other API besides healthcare that needs this but it is at least useful/needed for that API I believe. With how CallOptions currently work this is no way for a user to provide a key with multiple values.

File: googleapi/googleapi.go [hunk 1]
```diff
@@ -417,4 +417,23 @@ type traceTok string
...
+	key    string
+	values []string
+}
+
+// QueryParameter allows setting the value(s) of an arbitrary key.
+func QueryParameter(key string, values ...string) CallOption {
+	return queryParameter{key: key, values: append([]string{}, values...)}
+}
+
+// Get will never actually be called -- GetMulti will.
```
Comment (broady) [inline googleapi/googleapi.go:430]: This is a public package, I think this needs more documentation.

maybe add the type to googleapi:

```
type MultiCallOption interface {
   CallOption
   GetMulti() (string, []string)
}
```

then QueryParameter can return a MultiCallOption (or a better name, not sure)
and then document on MultiCallOption the difference in behavior.

Comment (codyoss) [reply googleapi/googleapi.go:438]: Done.

**Comment by broady** [comment]
LGTM (not sure why I can't leave an "Approval")

**Review by broady — APPROVED** [review]

**Pull request merged**